### PR TITLE
feat(discover) Add pagination and sorting to saved queries

### PR DESCRIPTION
**Pull request by markstory**
Enable saved queries to be sorted and paginated. This helps future proof the discover query list so that we don't have unbounded result sets in the future. I've included an escape hatch for discover1 which expects that it will receive *all* saved queries at once.

### CODE REVIEW

File: src/sentry/discover/endpoints/discover_saved_queries.py [hunk 1]
```diff
@@ -45,8 +44,32 @@ def get(self, request, organization):
...
+            paginator_cls = DateTimePaginator
+        elif sort_by in ("dateUpdated", "-dateUpdated"):
+            order_by = "-date_updated" if sort_by.startswith("-") else "date_updated"
+            paginator_cls = DateTimePaginator
+        else:
+            order_by = "name"
+            paginator_cls = OffsetPaginator
+
+        # Old discover expects all queries and uses this parameter.
+        if request.query_params.get("all") == "1":
```
Comment (dashed) [inline src/sentry/discover/endpoints/discover_saved_queries.py]: @markstory  To clarify, am I correct that the `all` query param can still be applied to Discover2+ queries whenever `query` is `'version:2'`?

Comment (markstory) [reply src/sentry/discover/endpoints/discover_saved_queries.py]: Yes. If you want to get all the queries and not paginate one could send this parameter.

File: tests/snuba/api/endpoints/test_discover_saved_query_detail.py [hunk 2]
```diff
@@ -65,7 +65,7 @@ def test_get_version(self):
 
         assert response.status_code == 200, response.content
         assert response.data["id"] == six.text_type(model.id)
-        assert response.data["projects"] == self.project_ids
+        assert set(response.data["projects"]) == set(self.project_ids)
```
Comment (dashed) [inline tests/snuba/api/endpoints/test_discover_saved_query_detail.py]: Classic.

**Review by dashed — APPROVED** [review]
I thought I already approved this. I'm approving now.

Looks good to me 👍

**Pull request merged**